pkg/kubectl/cmd/util/helpers.go

```diff
@@ -439,6 +440,9 @@ func Merge(codec runtime.Codec, dst runtime.Object, fragment, kind string) (runt
 	if err != nil {
 		return nil, err
 	}
+	if err = resource.ValidateSchema(patched, schema); err != nil {
```

Comment:
@sttts So, should we have something like
`if schema == nil {if err = resource.ValidateSchema(patched, schema); err != nil {...}
 }` ?